src/test/java/io/deepstream/message/ResubscriptionTest.java

```diff
@@ -0,0 +1,67 @@
+package io.deepstream.message;
+
+import io.deepstream.DeepstreamClient;
+import io.deepstream.constants.ConnectionState;
+import io.deepstream.utils.ResubscribeCallback;
+import io.deepstream.utils.ResubscribeNotifier;
+import io.socket.engineio.client.Socket;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+@RunWith( JUnit4.class )
+public class ResubscriptionTest {
+
+    String originalUrl = "originalProtocol://originalHost:originalPort";
+
+    Socket socketMock;
+    DeepstreamClient deepstreamClient;
+    ResubscribeCallback resubscribeCallbackMock;
+    ResubscribeNotifier resubscribeNotifier;
+    Connection connection;
+
+    @Before
+    public void setUp() throws URISyntaxException {
+        this.resubscribeCallbackMock = mock( ResubscribeCallback.class );

```

Comment:
Interesting... We would ideally mock out the deepstreamClient.. I'm wondering if creating an interface with what the client exposes to the rest of the classes makes sense

{
    onError()
   addConnectionStateListener()
   removeConnectionStateListener()
}

That way the tests would be alot easier. We could catch the one thats added and make sure it work. This will make us avoid having to create the client / socketMock and connection in alot of tests.